### PR TITLE
Fix some mem errors in 2024.09.1

### PR DESCRIPTION
**Pull request by ricrogz**
I hadn't done a mem tests run in a while.

Most of these are pointers leaking from tests, but there's also a couple of real things there, like the PDB parsing thing or the `combineMol()` one.

Besides these, there's also https://github.com/rdkit/rdkit/issues/7865, and some more leaks I hit in Inchi, but those can't be fixed here (I'm also not sure I build against the last release).

### CODE REVIEW

File: External/INCHI-API/inchi.cpp [hunk 1]
```diff
@@ -1246,18 +1246,32 @@ void cleanUp(RWMol &mol) {
...
+  char *m_data = nullptr;
+};
+
 }  // namespace
 
 RWMol *InchiToMol(const std::string &inchi, ExtraInchiReturnValues &rv,
                   bool sanitize, bool removeHs) {
   // input
-  char *_inchi = new char[inchi.size() + 1];
+  ManagedCharArray _inchi(inchi);
```
Comment (bp-kelley) [inline External/INCHI-API/inchi.cpp:1270]: I would just use

```
std::vector<char> _inchi(inchi.begin(), inchi.end()); 
```

As opposed to rolling a new class.

Comment (bp-kelley) [reply External/INCHI-API/inchi.cpp:1270]: Or you can use alloca and assume the stack isn't going to blow up.

Comment (ricrogz) [reply External/INCHI-API/inchi.cpp]: ... of course, it's never that easy, and we need a null terminator :P

File: Code/GraphMol/MolInterchange/Parser.cpp [hunk 2]
```diff
@@ -970,10 +970,13 @@ std::vector<boost::shared_ptr<ROMol>> DocToMols(
     }
     for (const auto &molval : doc["molecules"].GetArray()) {
       auto *mol = new RWMol();
+
+      // Add the mol to res, so that it is managed in case processMol throws
```
Comment (greglandrum) [inline Code/GraphMol/MolInterchange/Parser.cpp:974]: This changes behavior in the case that there is a problem
I'd prefer to handle the chance of a leak by making `mol` a `std::unique_ptr<RWMol>` and changing line 976 to:
`res.emplace_back(static_cast<ROMol *>(mol.release()))`

File: External/INCHI-API/inchi.cpp [hunk 3]
```diff
@@ -1253,11 +1253,14 @@ void cleanUp(RWMol &mol) {
 RWMol *InchiToMol(const std::string &inchi, ExtraInchiReturnValues &rv,
                   bool sanitize, bool removeHs) {
   // input
-  char *_inchi = new char[inchi.size() + 1];
+  std::vector<char> _inchi;
```
Comment (greglandrum) [inline External/INCHI-API/inchi.cpp:1256]: Why not just use an `std::string` and use its `data()` member?

**Review by greglandrum — CHANGES_REQUESTED** [review]
Thanks for doing this.

I made a few small suggestions

File: External/INCHI-API/inchi.cpp (hunk 3 above)
Comment (greglandrum) [reply External/INCHI-API/inchi.cpp:1256]: never mind... it returns a `const char *`

File: Code/GraphMol/MolInterchange/Parser.cpp (hunk 2 above)
Comment (ricrogz) [reply Code/GraphMol/MolInterchange/Parser.cpp:974]: Ok

**Comment by ricrogz** [comment]
I rebased on master so that we can run the CI after the commit that fixed the Mac builders.

**Review by greglandrum — APPROVED** [review]
LGTM

**Pull request merged**